[KHRGA-39] Updated EP id classes (#111)
KhronosGroup/SYCL_Reference: source/iface/id.rst

::

   template <int Dimensions = 1>
   class id;

> The ``id<int Dimensions>`` is a vector of Dimensions that is used

Reviewer: ```suggestion
The ``id`` class is a collection of size ``Dimensions`` that is used
```

Author: Fixed with : https://github.com/KhronosGroup/SYCL_Reference/pull/111/commits/3a9bce906a9622857614ce6806e543f7958ef6c8

::

  id();

  id(size_t dim0);

  id(size_t dim0, size_t dim1);

  id(size_t dim0, size_t dim1, size_t dim2);

  id(const range<Dimensions>& range);

  id(const item<Dimensions>& item);

Construct an ``id``.

An ``id`` can be 1, 2, or 3 dimensions. An ``id`` constructed from
a :ref:`range` uses the ``range`` values. An ``id`` constructed from
an :ref:`item` uses the ``id`` contained in the ``item``.

================
Member functions
================

``get``
=======

::

  size_t get(int dimension) const

> Return the value of the ``id`` for dimension ``Dimension``.

Reviewer: ```suggestion
Return the value of the ``id`` for dimension ``dimension``.
```

Author: Fixed with : https://github.com/KhronosGroup/SYCL_Reference/pull/111/commits/3a9bce906a9622857614ce6806e543f7958ef6c8

``<<=``, ``>>=``, ``&=``, ``|=``, ``^=``.

Assigns each element of ``lhs id`` instance with the result of an
element-wise ``OP`` operator between each element of ``lhs id`` and
each element of the ``rhs id`` and returns ``lhs id``.
If the operator returns a ``bool``, the result is the cast to ``size_t``.

::

  id& operatorOP(id& lhs, const size_t& rhs);

Where ``OP`` is: ``+=``, ``-=``, ``*=``, ``/=``, ``%=``,
``<<=``, ``>>=``, ``&=``, ``|=``, ``^=``.

Assigns each element of ``lhs id`` instance with the result of an
element-wise ``OP`` operator between each element of ``lhs id``
and the ``rhs size_t`` and returns ``lhs id``. If the operator
returns a ``bool``, the result is the cast to ``size_t``.

::

  id operatorOP(const size_t& lhs, const id& rhs);

Where ``OP`` is: ``+``, ``-``, ``*``, ``/``, ``%``, ``<<``,
``>>``, ``&``, ``|``, ``^``, ``&&``, ``||``, ``<``, ``>``,
``<=``, ``>=``.

Constructs and returns a new instance of the SYCL ``id`` class template
with the same dimensionality as the ``rhs`` SYCL ``id``, where each
element of the new SYCL ``id`` instance is the result of an element-wise
``OP`` operator between the ``lhs size_t`` and each element of the

> ``rhs`` SYCL ``ange``. If the operator returns a ``bool``,

Reviewer: ```suggestion
``rhs`` SYCL ``range``. If the operator returns a ``bool``,
```

Author: Fixed with : https://github.com/KhronosGroup/SYCL_Reference/pull/111/commits/3a9bce906a9622857614ce6806e543f7958ef6c8
